src/intervals/conversion.jl

```diff
@@ -17,7 +17,7 @@ convert(::Type{Interval{T}}, x::Bool) where {T} = convert(Interval{T}, Int(x))
 convert(::Type{Interval{T}}, x::Real) where {T} = atomic(Interval{T}, x)
 convert(::Type{Interval{T}}, x::T) where {T<:Real} = Interval{T}(x)
 convert(::Type{Interval{T}}, x::Interval{T}) where {T} = x
-convert(::Type{Interval{T}}, x::Interval) where {T} = atomic(Interval{T}, x)
+convert(::Type{Interval{T}}, x::Interval{S}) where {T,S} = atomic(Interval{T}, x)
 
 convert(::Type{Interval}, x::Real) = (T = typeof(float(x)); convert(Interval{T}, x))
 convert(::Type{Interval}, x::Interval) = x
```

Comment:
Not directly related to the issue, but this definition is the same as the one on line 20 and should probably be deleted (especially since the result would not be correct if for some weird reason it ever gets called).

Reply:
Sorry, I don't understand which definition you're talking about that is the same as the one on line 20?

Reply:
On line 23:

```jl
convert(::Type{Interval}, x::Interval) = x
```

Since `Interval` is the same as `Interval{T} where {T}` and nothing there requests both parameter to be the same.